tests/console/yast2_nis.pm

```diff
@@ -67,31 +67,42 @@ sub run() {
     send_key 'alt-y';
     wait_screen_change { type_string "-c" };                             # only checks if the config file has syntax errors and exits
     wait_screen_change { send_key 'alt-o' };
-    send_key_until_needlematch('nfs-client-configuration', 'alt-s', 2, 5);    # enter NFS configuration, sometimes fails, retry
-    send_key 'alt-a';                                                         # add nfs settings
-    assert_screen 'nfs-server-hostname';                                      # check that type string is sucessful
-    send_key 'alt-n';                                                         # from here enter some configurations...
+    wait_still_screen 3;
+    send_key 'alt-s';                                                    #NFS Configuration
+    wait_still_screen 3;
+
+    if (!check_screen 'nfs-client-configuration') {                      #If first alt-s is missed we send it again.
```

Comment:
you are just replacing `send_key_until_needlematch` with another way of retry ...
Instead of one line, now it is approx. 7 lines
You can just leave it and add `wait_still_screen 3;` in front of it

Reply:
You are right, but I think it can be done with `send_key_until_needlematch`, you can change timeouts and add wait functions

Reply:
yeah you are right, working on v2 which will basically wait for the next screen, also I think spamming alt-s is not the best solution, so replacing it with a simple assert_screen should do.

Reply:
You made it even better.

Reply:
Yeap, should work equal and it's easier to read.